docs/src/proposals/block_ancestors.md

```diff
@@ -0,0 +1,119 @@
+---
+title: Block Ancestors
+---
+
+There is no easy way to verify the parents of a given block just based on the
+shreds in the block. Having this information available in a block would be
+useful in scenarios where we want to reason about the structure of the chain
+without having to repair and replay all the ancestors. Such scenarios usually include
+collecting vote hashes from the cluster, and inferring cluster state like the weight
+of different forks based solely on those vote hashes, without needing to perform
+full repair/replay. 
+
+## Introducing ancestor proofs to a block
+
+Leaders will now include the last `N` ancestors proofs in a sorted list in
+the first shreds of their block `B`.
+
+Each item `A_i` in this ancestor proof list represents a block that is an ancestor
+of `B`. Each proof `A_i` proves that the block represented by `A_{i-1}` is its direct
+parent.
+
+The proof includes the following information about the ancestor block it represents:
+
+1) The slot `S_i`
+2) The block hash `blockhash_i` (i.e. the hash of the last tick in the block).
+3) The bank hash `bankhash_i`. Recall the bank hash looks like:
+```
+hash(&[
+    parent_bank_hash,
+    accounts_delta_hash,
+    &signature_count_buf,
+    last_blockhash,
+]
+```
+4) A merkle proof showing that `blockhash_i` is indeed the correct blockhash for `bankhash_i`. 
+From 3) we can see that this implies we need the `parent_bank_hash`, `accounts_delta_hash`, and`signature_count_buf`. Luckily, the `parent_bank_hash` should be available in `A_{i-1}`, for all
+but the last ancestor proof, so we can reuse that.
+
+Given the components of the proof, we now have the ability to for any block `A`
+for slot `S` that has an ancestor proof (recall a malicious leader may generate multiple
+blocks for a slot `S`):
+
+1) Tell whether we have a version of `S` that matches block `A`. This is because from the
+ancestor proof from `A`, we can find the expected blockhash. We can then check the last shred in
+our version of `S` for the hash of the last tick. If that checks out, we can then verify the hashes of all the entries in the shred are valid.
+
+2) Must be on the same fork as any other block `A'` for some slot `S'` that was also included in the
+proof list.
+
+## Removing EpochSlots from gossip
+
+EpochSlots takes a lot of bandwidth in gossip, and we would ultimately like to remove this
+burden on the network. Removing EpochSlots has a few fallouts described in the following
+sections.
+
+### Propagation Check
+
+TBD
```

Comment:
This is a major blocker, I'm not sure how we get a safe equivalent to the current propagation check without `EpochSlots`